### PR TITLE
Improve form usability & autosave

### PR DESCRIPTION
**Pull request by ohltyler**
### Description

Improves plugin usability in a few different ways:
- lower the autosave debounce amount from 10s -> 1s. Reason being, this minimizes the chances of updating a field, and it getting overwritten with some previous state. Long term, we may need to update how we handle autosave, as this could throttle or get expensive for making such often update workflow API calls. Worst case, we can fall back to manually saving.
- prevents unnecessary get workflow call after autosaving. the consequence is the "last updated" field is then not updated. Add TODO to handle at a later point.
- improved buttons to be disabled less often / more flexibility when navigating between forms
- gracefully handles running "run query" multiple times. The reprovision API will throw an error if there are no underlying template changes. We hide that error, and continue with executing the query.
- updates the presets - removes outdated ones, adds in a blank/custom one.
- adds TODOs for improving form state. Tracking issue w/ more details: #272 

### Issues Resolved
Makes progress on #23 

### Check List
- [x] Commits are signed per the DCO using `--signoff`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx [hunk 1]
```diff
@@ -676,6 +675,7 @@ export function WorkflowInputs(props: WorkflowInputsProps) {
                           onClick={() => {
                             validateAndRunIngestion();
                           }}
+                          // TODO: only enable if ingest is dirty
```
Comment (owaiskazi19) [inline public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx:678]: What do we mean here by "dirty"?

Comment (ohltyler) [reply public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx:678]: form is dirty / unsaved changes

Comment (owaiskazi19) [reply public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx:678]: Can we have a better name here? `Unsaved` or something

Comment (ohltyler) [reply public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx:678]: It's a common term for unsaved form data, and is used throughout `Formik` form library we use to maintain form and state. Prefer to leave as-is for consistency.

Comment (owaiskazi19) [reply public/pages/workflow_detail/workflow_inputs/workflow_inputs.tsx:678]: I see. Sounds good

**Review by owaiskazi19 — APPROVED** [review]

**Pull request merged**